test/unit/modules/delegates.js

```diff
@@ -489,6 +491,173 @@ describe('delegates', () => {
 				});
 			});
 		});
+
+		describe('getDelegateKeypairForCurrentSlot', () => {
+			let delegates;
+			let __private;
+
+			const genesis1 = {
+				passphrase:
+					'robust swift grocery peasant forget share enable convince deputy road keep cheap',
+				publicKey:
+					'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+			};
+
+			const genesis2 = {
+				passphrase:
+					'weapon van trap again sustain write useless great pottery urge month nominee',
+				publicKey:
+					'141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a',
+			};
+
+			const genesis3 = {
+				passphrase:
+					'course genuine appear elite library fabric armed chat pipe scissors mask novel',
+				publicKey:
+					'3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135',
+			};
+
+			let genesis1Keypair;
+			let genesis2Keypair;
+			let genesis3Keypair;
+
+			before(done => {
+				delegates = library.rewiredModules.delegates.__get__('self');
+				__private = library.rewiredModules.delegates.__get__('__private');
+
+				genesis1Keypair = lisk.getPrivateAndPublicKeyBytesFromPassphrase(
+					genesis1.passphrase
+				);
+				genesis2Keypair = lisk.getPrivateAndPublicKeyBytesFromPassphrase(
+					genesis2.passphrase
+				);
+				genesis3Keypair = lisk.getPrivateAndPublicKeyBytesFromPassphrase(
+					genesis3.passphrase
+				);
+
+				__private.keypairs = {};
+				__private.keypairs[genesis1.publicKey] = genesis1Keypair;
+				__private.keypairs[genesis2.publicKey] = genesis2Keypair;
+				__private.keypairs[genesis3.publicKey] = genesis3Keypair;
+
+				done();
+			});
+
+			it('should return genesis_1 keypair for slot N where (N % 101 === 35) in the first round', done => {
+				// For round 1, delegates genesis_1, genesis_2 and genesis_3 should forge for slots 35, 53 and 16 respectively.
+				const currentSlot = 35;
+				const round = 1;
+
+				delegates.generateDelegateList = (round, source, cb) => {
```

Comment:
This function is async, we're not waiting for it so there is no guarantee that it will finish before the next one. Globally in those tests.

Reply:
I'm overriding the `generateDelegateList` function and not executing it.
It will be executed during `getDelegateKeypairForCurrentSlot` call.

Reply:
Aaaa, ok. 😄